### PR TITLE
Implemented InfluxDB for Logging Purposes

### PR DESCRIPTION
**Pull request by Antonwy**
## ⚠️ Merge Order
- Merge after #100 

## Description
- Implemented `InfluxDB` connection
- Added InfluxDB testing environment to docker compose
- Can be used to log events => events can then be analysed in the frontend ui which can be found at `localhost:8086` 
- Username and Password can be specified in the `.env` file

## Usage
This PR does not use InfluxDB it just does the connection. #100 uses Influx to log some metrics about the scheduling.

**Example usage:**
```go
func LogFileDownload() {
	p := influxdb2.NewPointWithMeasurement("file_download").
		AddTag("user", "test").
		AddField("file", "test")
	LogPoint(p)
}
```
⚠️ Caution, `write.Flush()` will instantly store the log in the db which can be performance heavy.
Without `write.Flush()` the Influx Client will batch save all logs after **5000** entries.

I have set the `batchSize` to **1000** per default.

### CODE REVIEW

File: server/backend/influxdb.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	// This is to reduce the number of requests to the database
+	InfluxBatchSize = 1_000
+)
+
+var (
+	influxToken = os.Getenv("INFLUXDB_TOKEN")
+	influxURL   = os.Getenv("INFLUXDB_URL")
+)
+
+func ConnectToInfluxDB() error {
```
Comment (joschahenningsen) [inline server/backend/influxdb.go:22]: Seems like all this could go into the influx package, no?

File: server/backend/influxdb.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	if influxToken == "" {
+		return errors.New("no influxdb token provided")
```
Comment (joschahenningsen) [inline server/backend/influxdb.go:24]: imho if influx is not configured, incoming metrics should be silently dropped.

File: server/backend/influxdb.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	"errors"
+	"github.com/TUM-Dev/Campus-Backend/server/backend/influx"
+	influxdb2 "github.com/influxdata/influxdb-client-go/v2"
+	"os"
+)
+
+const (
+	// InfluxBatchSize is the number of points that are sent to influxdb at once
+	// This is to reduce the number of requests to the database
+	InfluxBatchSize = 1_000
```
Comment (joschahenningsen) [inline server/backend/influxdb.go:14]: 1000 seems like a lot, how many datapoints do you expect in production (e.g. per minute)?

File: server/backend/influxdb.go (hunk 1 above)
Comment (Antonwy) [reply server/backend/influxdb.go:22]: I think I can't access the influx package from the main file because it is in the backend package.

**Comment by COM8** [comment]
A more general question: Why is this needed at all? What speaks against using text logs for this functionality?
Yes, I see the point with a DB is easier to query and so on. But in my eyes this is something that can be archived with structured logs as well. Perhaps add a [DEVICE-ID] log tag or something like this.

Adding support for this in my eyes adds unnecessary extra complexity.

**Comment by joschahenningsen** [comment]
Observability is always great, since we already have influxdb monitoring for http requests running on the prod server this seems like a nice idea imo. As long as it doesn't disrupt local development without influx of course.

**Comment by COM8** [comment]
OK, if it already exists, why not. Thanks @joschahenningsen 👍🏻

File: .env [hunk 4]
```diff
@@ -1,4 +1,12 @@
...
\ No newline at end of file
+
+ENVIRONMENT=dev
+
+INFLUXDB_USER=campus_backend
+INFLUXDB_PASSWORD=secret_influxdb_password
+INFLUXDB_ORG=tum
+INFLUXDB_BUCKET=campus_backend
+INFLUXDB_URL=http://influxdb:8086
+INFLUXDB_ADMIN_TOKEN=secret_influxdb_admin_token
```
Comment (COM8) [inline .env:12]: Bight be worth documenting those here?
https://github.com/TUM-Dev/Campus-Backend#environment-variables

**Review by COM8 — CHANGES_REQUESTED** [review]
Just a bit of documentation inside the README.md and we are ready to go here as well 🎉 !

**Review by COM8 — APPROVED** [review]

**Pull request merged**